satpy/writers/__init__.py

```diff
@@ -180,17 +180,24 @@ def _determine_mode(dataset):
 
 def add_overlay(orig, area, coast_dir, color=(0, 0, 0), width=0.5, resolution=None,
                 level_coast=1, level_borders=1, fill_value=None):
-    """Add coastline and political borders to image, using *color* (tuple
-    of integers between 0 and 255).
-    Warning: Loses the masks !
-    *resolution* is chosen automatically if None (default), otherwise it should be one of:
+    """Add coastline and political borders to image.
+
+    Uses ``color`` (tuple of integers between 0 and 255).
```

Comment:
I'd like to have the possibility to keep the image in `L` mode of the colour provide just has one value, eg `add_overlay(..., ..., ..., color=(255,), ...)` would just add white coastlines and keep the image in `L` mode

Reply:
also, maybe specify that the tuple is expected to be in rgb if the length is 3, etc

Reply:
I can add the documentation. I agree that the `L` mode preservation would be nice but that is something that needs to be supported by `pycoast`, right? Or I guess I could check every color that was passed by the user (not easy to do for the decorations) and convert back from RGB to L? I don't really want to do that. 😄

Reply:
we can keep it for later